backend/tracim_backend/lib/cleanup/cleanup.py

```diff
@@ -0,0 +1,339 @@
+        self.safe_delete(workspace)
+        return workspace_id
+
+    def delete_workspace_agenda(self, workspace_id: int) -> typing.Optional[str]:
+        # INFO - G.M - 2019-12-11 - delete workspace agenda
+        if not self.app_config.CALDAV__ENABLED:
+            logger.info(
+                self,
+                "will not delete workspace {} agenda, because caldav feature is disabled".format(
+                    workspace_id
+                ),
+            )
+            return
+
+        agenda_dir = "{}{}{}{}".format(
+            self.app_config.CALDAV__RADICALE__STORAGE__FILESYSTEM_FOLDER,
+            "/collection-root",
+            self.app_config.CALDAV_RADICALE_WORKSPACE_PATH,
+            workspace_id,
+        )
+        logger.info(self, "delete workspace {} agenda dir : {}".format(workspace_id, agenda_dir))
+        self.safe_delete_dir(agenda_dir)
+        return agenda_dir
+
+    def delete_user_associated_data(self, user: User) -> None:
+        """deleted all stuff about user but user itself, his workspaces and content"""
+
+        # INFO - G.M - 2019-12-11 - user permission
+        upload_permissions = self.session.query(UploadPermission).filter(
+            UploadPermission.author_id == user.user_id
+        )
+        for upload_permission in upload_permissions:
+            logger.info(
+                self,
+                "delete upload_permission {} from user {}".format(
+                    upload_permission.upload_permission_id, upload_permission.author_id
+                ),
+            )
+            self.safe_delete(upload_permission)
+
+        # INFO - G.M - 2019-12-11 - user share
+        shares = self.session.query(ContentShare).filter(ContentShare.author_id == user.user_id)
+        for share in shares:
+            logger.info(self, "delete share {} from user {}".format(share.share_id, user.user_id))
+            self.safe_delete(share)
+
+        # INFO - G.M - 2019-12-11 - User role
+        roles = self.session.query(UserRoleInWorkspace).filter(
+            UserRoleInWorkspace.user_id == user.user_id
+        )
+        for role in roles:
+            logger.info(
+                self,
+                "delete role for user {} of workspace {}".format(role.user_id, role.workspace_id),
+            )
+            self.safe_delete(role)
+
+        logger.info(self, "delete user groups for user {}".format(user.user_id))
+        user.groups = []
+        self.safe_update(user)
+
+    def delete_user_agenda(self, user_id: int) -> typing.Optional[str]:
+        """
+        delete agenda of user
+        :param user_id: user_id of user whe delete agenda
+        :return: path of deleted agenda
+        """
+
+        if not self.app_config.CALDAV__ENABLED:
+            logger.info(
+                self,
+                "will not delete user {} agenda, because caldav feature is disabled".format(
+                    user_id
+                ),
+            )
+            return
```

Comment:
idem (I'm not sure about this. We should delete agenda user data even if radicale disabled.)

Reply:
done.